vmdb/db/fixtures/ae_datastore/ManageIQ/Infrastructure/VM/Retirement/StateMachines/Methods.class/__methods__/remove_from_provider.rb

```diff
@@ -0,0 +1,30 @@
+#
+# Description: This method removes the VM from the provider
+#
+
+# Get vm from root object
+vm = $evm.root['vm']
+category = "lifecycle"
+tag = "retire_full"
+
+removal_type = $evm.inputs['removal_type'].downcase
+$evm.set_state_var('vm_removed_from_provider', false)
+
+if vm
+  ems = vm.ext_management_system
+  case removal_type
+  when "remove_from_disk"
+    if vm.miq_provision || vm.tagged_with?(category, tag)
+      $evm.log('info', "Removing VM:<#{vm.name}> from provider:<#{ems.try(:name)}>")
+      vm.remove_from_disk
+      $evm.set_state_var('vm_removed_from_provider', true)
```

Comment:
@mkanoor 
I think remove_from_provider is more descriptive because it encompasses the removal type of unregister and matches the state machine state and method naming.